lib/validators/specValidator.ts

```diff
@@ -220,31 +223,38 @@ export class SpecValidator<T extends CommonValidationResult> {
    *
    * @return {object} err Return the constructed Error object.
    */
-  protected constructErrorObject<TE extends CommonError>(
+  protected readonly constructErrorObject = <TE extends CommonError>(
     code: ErrorCode,
     message: string,
     innerErrors?: null | TE[],
-    skipValidityStatusUpdate?: boolean
-  ): TE {
+    skipValidityStatusUpdate?: boolean,
+    source?: json.JsonObject
```

Comment:
nit: can we make a params object so we avoid passing in `undefined` ? also would make it easier to see what the params are when used since there are a lot of them